pkgs/tools/networking/networkmanager/default.nix

```diff
@@ -145,15 +146,18 @@ stdenv.mkDerivation rec {
     ppp
     libndp
     curl
-    mobile-broadband-provider-info
+    jansson
+    dbus # used to get directory paths with pkg-config during configuration
+  ] ++ lib.optionals (!libnmOnly) [
+    # for nmtui
+    newt
+    readline
+    # for plugins
     bluez5
     dnsmasq
-    modemmanager
-    readline
-    newt
     libsoup
-    jansson
-    dbus # used to get directory paths with pkg-config during configuration
+    mobile-broadband-provider-info
+    modemmanager
```

Comment:
I moved most plugin dependencies out here, so build and runtime closure size should both be significant smaller now.